tools/c7n_gcp/tests/test_compute.py

```diff
@@ -108,14 +108,38 @@ def test_delete_instance(self):
 class DiskTest(BaseTest):
 
     def test_disk_query(self):
-        factory = self.replay_flight_data('disk-query')
+        factory = self.replay_flight_data('disk-query', project_id='custodian-1291')
         p = self.load_policy(
             {'name': 'all-disks',
              'resource': 'gcp.disk'},
             session_factory=factory)
         resources = p.run()
         self.assertEqual(len(resources), 6)
 
+    def test_disk_snapshot(self):
+        factory = self.replay_flight_data('disk-snapshot', project_id='custodian-1291')
+        p = self.load_policy(
+            {'name': 'all-images',
+             'resource': 'gcp.disk',
+             'filters': [
+                 {'name': 'c7n-jenkins'}],
+             'actions': ['snapshot']},
+            session_factory=factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+
+    def test_disk_delete(self):
+        factory = self.replay_flight_data('disk-delete', project_id='custodian-1291')
+        p = self.load_policy(
+            {'name': 'all-images',
+             'resource': 'gcp.disk',
+             'filters': [
+                 {'name': 'c7n-jenkins'}],
+             'actions': ['delete']},
+            session_factory=factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
```

Comment:
so for a delete we typically construct a client `client = p.resource_manager.get_client()` and query that the resource is no longer present, ie verify the effect of the action  via get/list query for the resource.